tools/docker-network/run.sh

```diff
@@ -23,8 +23,15 @@ fi
 export DOCKER_BUILDKIT=1
 export COMPOSE_DOCKER_CLI_BUILD=1
 echo "Build iota-core"
+
 # Allow docker compose to build and cache an image
-docker compose -f $DOCKER_COMPOSE_FILE build --build-arg WITH_GO_WORK=${WITH_GO_WORK:-0}
+if [[ "$WITH_GO_WORK" -eq 1 ]]
+then
+  export DOCKER_BUILD_CONTEXT="../../../"
+  export DOCKERFILE_PATH="./iota-core/Dockerfile"
+fi
```

Comment:
Would be a bit easier to understand if you declare all defaults before this if and then override. Then you just need to pass it in the call to `docker compose`